src/docs/upgrading.dox

```diff
@@ -1,5 +1,67 @@
 /*! @page upgrading Upgrading WiredTiger applications
+@section version_310 Upgrading to Version 3.1.0
+<dl>
+
+<dt>WiredTiger on-disk log file format change</dt>
+<dd>
+The WiredTiger on-disk file format for write-ahead log files has changed
+as the log file version number was incremented. See
+<a href=https://jira.mongodb.org/browse/WT-4029>WT-4029</a> for details.
+</dd>
+
+<dt>::wiredtiger_open compatibility configuration changes</dt>
+<dd>
+The compatibility setting now takes additional options that can define
+the minimum or maximum required version of existing data files. See
+<a href=https://jira.mongodb.org/browse/WT-4098>WT-4098</a> for details.
+</dd>
+
+<dt>Changed transaction semantics around schema operations</dt>
+<dd>
+WiredTiger does not offer fully transactional create and drop operations.
+We have made some changes to how create and drop are implemented
+if done within the scope of an explicit transaction. If an application
+is relying on particular visibility/atomicity guarantees around table
+create or drop, care should be taken when upgrading. See
+<a href=https://jira.mongodb.org/browse/WT-3964>WT-3964</a> for details.
+</dd>
+
+<dt>On-disk format change for metadata</dt>
+<dd>
+There was a change to the content stored in the WiredTiger owned metadata
+files, which means metadata created or updated by this version of WiredTiger
+is not compatible with earlier versions. See
+<a href=https://jira.mongodb.org/browse/WT-3905>WT-3905</a> for details.
+</dd>
+
+<dt>::wiredtiger_open eviction_checkpoint_target configuration change</dt>
+<dd>
+The default value for the \c eviction_checkpoint_target configuration
+setting was changed from 5 to 1. This was due to a change in the algorithm
+to focus more on reducing dirty content in low throughput workloads. See
+<a href=https://jira.mongodb.org/browse/WT-4111>WT-4111</a> for details.
```

Comment:
Please note WT-4111 was backported to MongoDB v4.0 very recently. If we'd line up WiredTiger contents between 3.1.0 and MongoDB 4.0.0, then WT-4111 should be excluded.